addons/account/views/account_payment_view.xml

```diff
@@ -267,18 +267,18 @@
                                 <field name="payment_method_line_id" required="1" options="{'no_create': True, 'no_open': True}"
                                        attrs="{'readonly': [('state', '!=', 'draft')]}"/>
 
-                                <field name="partner_bank_id" context="{'default_partner_id': partner_id}" string="Customer Bank Account"
+                                <field name="partner_bank_id" context="{'default_partner_id': partner_id}" string="Recipient Bank Account"
                                         attrs="{
                                             'invisible': ['|', '|', ('show_partner_bank_account', '=', False), ('partner_type','!=','customer'), ('is_internal_transfer', '=', True)],
                                             'required': [('require_partner_bank_account', '=', True), ('is_internal_transfer', '=', False)],
-                                            'readonly': [('state', '!=', 'draft')]
+                                            'readonly': ['|', ('state', '!=', 'draft'), ('payment_type', '=', 'inbound')],
                                         }"/>
 
-                                <field name="partner_bank_id" context="{'default_partner_id': partner_id}" string="Vendor Bank Account"
+                                <field name="partner_bank_id" context="{'default_partner_id': partner_id}" string="Recipient Bank Account"
                                         attrs="{
                                             'invisible': ['|', '|', ('show_partner_bank_account', '=', False), ('partner_type','!=','supplier'), ('is_internal_transfer', '=', True)],
                                             'required': [('require_partner_bank_account', '=', True), ('is_internal_transfer', '=', False)],
-                                            'readonly': [('state', '!=', 'draft')]
+                                            'readonly': ['|', ('state', '!=', 'draft'), ('payment_type', '=', 'inbound')],
```

Comment:
I discussed these changes with tsb, and we'll do it a little differently, still keeping the old label in the cases where they are legit.

As it's kind of a bug, I made it in stable here: https://github.com/odoo/odoo/pull/83674